src/__tests__/labs/Labs.test.tsx

```diff
@@ -1,84 +1,62 @@
 import { RootState } from '../../shared/store'
 
 const { TitleProvider } = titleUtil
 const mockStore = createMockStore<RootState, any>([thunk])
 
 describe('Labs', () => {
-  jest.spyOn(titleUtil, 'useUpdateTitle').mockImplementation(() => jest.fn())
-  jest.spyOn(LabRepository, 'findAll').mockResolvedValue([])
-  jest
-    .spyOn(LabRepository, 'find')
-    .mockResolvedValue({ id: '1234', requestedOn: new Date().toISOString() } as Lab)
-  jest
-    .spyOn(PatientRepository, 'find')
-    .mockResolvedValue({ id: '12345', fullName: 'test test' } as Patient)
-
-  const setup = async (initialEntry: string, permissions: Permissions[] = []) => {
+  const setup = (initialEntry: string, permissions: Permissions[] = []) => {
     const store = mockStore({
       user: { permissions },
     } as any)
 
-    let wrapper: any
-    await act(async () => {
-      wrapper = await mount(
-        <Provider store={store}>
-          <MemoryRouter initialEntries={[initialEntry]}>
-            <TitleProvider>
-              <Labs />
-            </TitleProvider>
-          </MemoryRouter>
-        </Provider>,
-      )
-    })
-
-    wrapper.update()
-    return { wrapper: wrapper as ReactWrapper }
+    return render(
+      <Provider store={store}>
+        <MemoryRouter initialEntries={[initialEntry]}>
+          <TitleProvider>
+            <Labs />
+          </TitleProvider>
+        </MemoryRouter>
+      </Provider>,
+    )
   }
 
   describe('routing', () => {
     describe('/labs/new', () => {
       it('should render the new lab request screen when /labs/new is accessed', async () => {
-        const { wrapper } = await setup('/labs/new', [Permissions.RequestLab])
+        const { container } = setup('/labs/new', [Permissions.RequestLab])
 
-        expect(wrapper.find(NewLabRequest)).toHaveLength(1)
+        expect(container.querySelector('form')).toBeInTheDocument()
       })
 
       it('should not navigate to /labs/new if the user does not have RequestLab permissions', async () => {
-        const { wrapper } = await setup('/labs/new')
+        const { container } = setup('/labs/new')
 
-        expect(wrapper.find(NewLabRequest)).toHaveLength(0)
+        expect(container.querySelector('form')).not.toBeInTheDocument()
       })
     })
 
     describe('/labs/:id', () => {
       it('should render the view lab screen when /labs/:id is accessed', async () => {
-        const { wrapper } = await setup('/labs/1234', [Permissions.ViewLab])
+        setup('/labs/1234', [Permissions.ViewLab])
 
-        expect(wrapper.find(ViewLab)).toHaveLength(1)
+        expect(screen.getByText(/loading/i)).toBeInTheDocument()
       })
     })
 
     it('should not navigate to /labs/:id if the user does not have ViewLab permissions', async () => {
-      const { wrapper } = await setup('/labs/1234')
+      setup('/labs/1234')
 
-      expect(wrapper.find(ViewLab)).toHaveLength(0)
+      expect(screen.queryByText(/loading/i)).not.toBeInTheDocument()
```

Comment:
I'm worried that this test may not fail in the future if the loading message is ever changed and the permission logic becomes broken. 

In other words, I don't think that the absence of the a loading spinner indicates that the permissions logic is working properly. I wonder if we can assert on the route or something other than "loading..." in the dom.

Reply:
@nobrayner

Reply:
@JacobMGEvans This has been resolved